browser/widevine/BUILD.gn

```diff
@@ -84,3 +88,66 @@ source_set("browser_tests") {
     "//url",
   ]
 }
+
+if (is_win) {
+  sign("signed_brave_exe") {
+    sources = [ "$root_out_dir/brave.exe" ]
+    outputs = [ "$root_out_dir/signed_binaries/brave.exe" ]
+    deps = [ "//brave/build/win:copy_exe" ]
+  }
+
+  sign("signed_chrome_dll") {
+    sources = [ "$root_out_dir/chrome.dll" ]
+    outputs = [ "$root_out_dir/signed_binaries/chrome.dll" ]
```

Comment:
if we keep this double-signing setup for now, then how about changing the dir name to reflect things better? Like `widevine_signed_binaries`. It will make things much clearer in `SignAndCopyPreSignedBinaries`.

Currently `signed_binaries` name makes you think that it's always created when signing is enabled, but in reality this dir is only created when sign && widevine is enabled.

Reply:
I'm up for improving the name but `signed_binaries` isn't just created when `sign && enable_widevine`. It's actually created when `sign`. Specifically, `setup.exe` is placed in `signed_binaries/` independently of `enable_widevine`.

Reply:
> I'm up for improving the name but `signed_binaries` isn't just created when `sign && enable_widevine`. It's actually created when `sign`. Specifically, `setup.exe` is placed in `signed_binaries/` independently of `enable_widevine`.

how `signed_binaries/setup.exe` is used during the build? is it picked up by CI at some point? Or it should be used by other gn targets? I don't see where the signed file is referenced.

If we will keep `signed_binaries`, then I suggest to not sign `brave.exe/chrome.dll` via two different code paths. These are two most important binaries, they definitely shouldn't be created by `widevine:*` targets in one of the build scenarios. The right thing here is to have a single unconditional code-path for `brave.exe/chrome.dll` sign somewhere in the gn dependency.

one of options:
1. move `brave.exe`, `chrome.dll` sign into the main build process and make it independent of `widevine/cdm_host_verification` gn flags.
2. skip `brave.exe`, `chrome.dll` sign by adding an explicit filter to `sign_binaries(staging_dir)` call in `create_installer_archive.py`.
3. always use these two pre-signed binaries in `create_installer_archive.py` and remove all mentions of `enable_widevine` there.

Reply:
> how signed_binaries/setup.exe is used during the build?

[Upstream's PrepareSetupExec](https://source.chromium.org/chromium/chromium/src/+/main:chrome/tools/build/win/create_installer_archive.py;drc=acc7c7df955646c8a391c5e60d61eed858cd70b1;l=366) takes `setup.exe` from `options.build_dir`. My PR [overwrites this](https://github.com/brave/brave-core/pull/18043/files#diff-6d23e1ce6aeeb5c22a61f808f78557cb44a353371a7eb081fd7e183a0ce4a45aR90) so `options.build_dir` becomes `signed_binaries/`. The effect is that, when upstream's code runs to generate the mini_installer, the signed `setup.exe` gets used.

Reply:
@goodov by option 3., do you essentially mean the following?

```
-def SignAndCopyPreSignedBinaries(skip_signing, enable_widevine, output_dir,
-                                 staging_dir, current_version):
+def SignAndCopyPreSignedBinaries(skip_signing, output_dir, staging_dir,
+                                 current_version):
     if not skip_signing:
         sign_binaries(staging_dir)
-        if enable_widevine:
-            # Copy files that already were signed into the staging directory.
-            # This is important to make sure that their associated .sig files,
-            # which contain their signature, remain valid.
-            src_dir = os.path.join(output_dir, 'signed_binaries')
-            chrome_dir = os.path.join(staging_dir, CHROME_DIR)
-            version_dir = os.path.join(chrome_dir, current_version)
-            shutil.copy(os.path.join(src_dir, 'brave.exe'), chrome_dir)
-            shutil.copy(os.path.join(src_dir, 'chrome.dll'), version_dir)
+        # Copy pre-signed files into the staging directory. This is important
+        # for when Widevine is enabled.
+        src_dir = os.path.join(output_dir, 'signed_binaries')
+        chrome_dir = os.path.join(staging_dir, CHROME_DIR)
+        version_dir = os.path.join(chrome_dir, current_version)
+        shutil.copy(os.path.join(src_dir, 'brave.exe'), chrome_dir)
+        shutil.copy(os.path.join(src_dir, 'chrome.dll'), version_dir)
```

Reply:
Ah and we could rename `signed_binaries/` to `presigned_binaries/`.

Reply:
> by option 3., do you essentially mean the following?

yes.

> Ah and we could rename `signed_binaries/` to `presigned_binaries/`.

that would also be nice.

Reply:
I've renamed `signed_binaries` to `presigned_binaries` in 5361ad1ede09c4dd8e3abe8990297beba2315975 and implemented option 3 in abe4998ef3be3f69344a34f4ee39c3f2be846919.